app/test_main.py

```diff
@@ -1 +1,27 @@
-# write your code here
+import pytest
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "is_valid,has_connection,result",
+    [
+        (True, True, "Accessible"),
+        (False, False, "Not accessible"),
+        (False, True, "Not accessible"),
+        (True, False, "Not accessible")
+    ]
+)
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
+def test_can_access_google_page(
+        mock_valid_google_url: bool,
+        mock_has_internet_connection: bool,
```

Comment:
`mock_valid_goodle_url` and `mock_has_internet_connection` are not bools. fix the type hint.